components/menu/menu-item.js

```diff
@@ -0,0 +1,45 @@
+import '../icons/icon.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { MenuItemMixin } from './menu-item-mixin.js';
+import { menuItemStyles } from './menu-item-styles.js';
+
+class MenuItem extends MenuItemMixin(LitElement) {
+
+	static get styles() {
+		return [ menuItemStyles,
+			css`
+				:host {
+					display: flex;
+					padding: 0.75rem 1rem;
+				}
+
+				:host > span {
+					flex: auto;
+					line-height: 1rem;
+					overflow-x: hidden;
+					overflow-y: hidden;
+					text-overflow: ellipsis;
+					white-space: nowrap;
+				}
+
+				:host > d2l-icon {
+					flex: none;
+					margin-top: 0.1rem;
+				}
+			`
+		];
+	}
+
+	render() {
+		const icon = this.hasChildView ?
+			html`<d2l-icon icon="tier1:chevron-right"></d2l-icon>` : '';
```

Comment:
Empty string will probably work, but wondering what the best practice here should be for when we don't want a view. Maybe `null`?

Reply:
Yeah `null` and `undefined` will work fine as well... I've been using `null`.